### PR TITLE
[IdentityLookup] Adds Xcode 9 Beta 1 bindings

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

**Comment by dalexsoto** [comment]
Keeping this as reference https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/4129/

**Comment by dalexsoto** [comment]
build

File: src/identitylookup.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject))]
+	interface ILMessageFilterExtension {
+	}
+
+	[iOS (11,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSExtensionContext))]
+	interface ILMessageFilterExtensionContext {
+
```
Comment (timrisi) [inline src/identitylookup.cs]: Whitespace

File: src/identitylookup.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		[Async]
+		void DeferQueryRequestToNetwork (Action<ILNetworkResponse, NSError> completion);
+	}
+
+	interface IILMessageFilterQueryHandling { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface ILMessageFilterQueryHandling {
+
```
Comment (timrisi) [inline src/identitylookup.cs]: Same (and others after)

File: src/identitylookup.cs (hunk 1 above)
Comment (dalexsoto) [inline src/identitylookup.cs]: I always leave a Whitespace between the interface line definition and the first member definition and this is the first time I've been called out :)

File: src/identitylookup.cs (hunk 2 above)
Comment (dalexsoto) [inline src/identitylookup.cs]: I always leave a Whitespace between the interface line definition and the first member definition and this is the first time I've been called out :)

File: src/identitylookup.cs (hunk 1 above)
Comment (timrisi) [inline src/identitylookup.cs]: Probably not a big deal either way, and I don't think there's a policy for it.  I never leave the whitespace there, and glancing through a couple of the larger binding files most interfaces seemed to not have it, though looking elsewhere there's plenty that DO have it, so...¯\_(ツ)_/¯

**Comment by dalexsoto** [comment]
@timrisi see for example intents https://github.com/xamarin/xamarin-macios/blob/master/src/intents.cs do you have a reference to our coding guidelines where I shouldn't do so?

**Comment by timrisi** [comment]
Don't know of any guidelines.  Foundation mostly seems to not have it, so I this is probably one case with no current guideline

**Comment by dalexsoto** [comment]
unrelated failure
* Net.Http.NSUrlSessionHandler) -> https://bugzilla.xamarin.com/show_bug.cgi?id=57762

File: src/identitylookup.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		[Abstract]
+		[Export ("handleQueryRequest:context:completion:")]
+		void HandleQueryRequest (ILMessageFilterQueryRequest queryRequest, ILMessageFilterExtensionContext context, Action<ILMessageFilterQueryResponse> completion);
```
Comment (spouliot) [inline src/identitylookup.cs]: that's repeating `QueryRequest` from the first parameter 
but `Handle` is not very descriptive and already used for a property
can you just double-check the swift name ?

Comment (dalexsoto) [inline src/identitylookup.cs]: @spouliot they use `handle` in swift `func handle(ILMessageFilterQueryRequest, context: ILMessageFilterExtensionContext, completion: (ILMessageFilterQueryResponse) -> Void)` but `Handle` on our side has a very special meaning and also can conflict if implemented by a NSObject which will cause additional pain (code) on the users end. Also for readability `HandleQueryRequest` IMHO is much more descriptive.

I think we can bend our rules a little this time but it is your call 👍

Comment (spouliot) [inline src/identitylookup.cs]: Thanks, I agree - let's merge that puppy.

**Pull request merged**